config/configgates/README.md

```diff
@@ -0,0 +1,92 @@
+# Collector Feature Gates
+
+This package provides a mechanism that allows operators to enable and disable
+experimental or transitional features at deployment time. These flags should
+be able to govern the behavior of the application starting as early as possible
+and should be available to every component such that decisions may be made
+based on flags at the component level.
+
+## Usage
+
+A component that needs a feature gate that will not be shared with other 
+components should define a `Gate` as an unexported package variable and add 
```

Comment:
Is there anything that prevents to export the variable and use it directly in other _related_ packages if needed (other than the danger of direct coupling of different packages)?

Reply:
I don't think I'd want to have to worry about policing such uses.  In any event, the refactoring prompted by discussion with Bogdan above renders this pattern unusable as the `Registry` only holds the default values for registered gates and values applied from configuration are only visible in the `frozenRegistry` implementation of the `Gates` interface made available to a `Service` through `config.Config`.  This means that anything that wishes to know the configured state of a `Gate` must use the `Gates` interface.